### ISSUE TITLE
Issue with Reddit and Twitch

### ISSUE THREAD

**Issue by verlain3** [ghacksuserjs/ghacks-user.js#78]
Hello, 

Since version 0.11 i am unable to login on reddit. Do you have any idea which line from the user.js might be interfering with this?

Also, on Twitch i am unable to watch any stream. 
Thanks for any help

**Comment by Thorin-Oakenpants** [comment]
Login: cookies are disabled by default .. in v52 this is number 2701 `network.cookie.cookieBehavior`. We would expect users to have a cookie add-on by now

Twitch streams: I don't do twitch, but look at items 1820 (GMP) +1825(Widevine) + 1830(EME) + 1840(OpenH264) as these all have to do with video

**Comment by earthlng** [comment]
just tested v11 in a fresh FF52.0.2 profile without any addons and for twitch to work you need to change the following prefs:

javascript.options.asmjs;true -> probably required
network.cookie.cookieBehavior;0 -> somehow enable cookies
dom.indexedDB.enabled;true -> required
media.autoplay.enabled;true -> required 
media.block-autoplay-until-in-foreground;false ->maybe not required

if you watch a lot of twitch you might want to enable hardware acceleration to save some power, but it works without those as well
user_pref("gfx.direct2d.disabled", true);
user_pref("layers.acceleration.disabled", true);

**Comment by verlain3** [comment]
Thanks Thorin-Oakenpants & earthlng!

It works just by enabling media.autoplay.enabled to true and by setting network.cookie.cookieBehavior to 1 -  only accept from the originating site (block third party cookies).

One more question though. Here on Github, the user.js version is different (and newer) from ghacks website, right? On ghacks it's recommended for version 51 of Firefox and here there's work being made for 52 version and up, am i correct?

**Comment by Thorin-Oakenpants** [comment]
@verlain3 . after v11 was updated to ghacks.net (and updates to there were basically 6 monthly), it was decided to come here. Since github obviously allows us to make changes in real time in public etc .. this is what we do.

- We changed our versioning numbers to match Firefox stable, so basically version 11 is version 51
- The user.js (the live one that we make commits to .. [here](https://github.com/ghacksuserjs/ghacks-user.js/blob/master/user.js), or when you download a [zip](https://github.com/ghacksuserjs/ghacks-user.js/archive/master.zip)) is always up to date as per the version number in the js.
- We do [releases](https://github.com/ghacksuserjs/ghacks-user.js/releases) and [here](https://github.com/ghacksuserjs/ghacks-user.js/issues/54) you can read all about how those

So basically, we know in advance a lot of the changes coming to the next FF stable (see our diffs issues like [this one](https://github.com/ghacksuserjs/ghacks-user.js/issues/47)).  As soon a new stable lands from Firefox, we make a release of the "old" version and as fast as we can get the live "master" user.js up to speed (from 51 to 52 we were about 6 days after stable landed). When we finish these (that is adding all the changes to a new stable), we post a changelog (like this one: https://github.com/ghacksuserjs/ghacks-user.js/issues/53 ) and Martin does an article at ghacks (What's new in ghacks user.js 53).

And yes, this version is miles ahead of the ghacks article.

**Comment by earthlng** [comment]
You're welcome.

Thanks for reporting back (I changed my comment for `dom.indexedDB.enabled`)

Yes and Yes to your questions. [This](https://github.com/ghacksuserjs/ghacks-user.js/issues/65#issuecomment-290703628) could help if you want to move to the latest version of this user.js

**Comment by earthlng** [comment]
Damn Pants, always a moment faster aren't you?

**Comment by Thorin-Oakenpants** [comment]
SpeedyPants. PS: did the wiki page

**Comment by verlain3** [comment]
Much appreciated, Thorin-Oakenpants & earthlng. Valuable information and great work done here. I can only wish you keep doing this for the Firefox community.

Thank you very much.

**Comment by verlain3** [comment]
This is not working anymore, apparently. Strange thing is, if i use alternatives twitch players such as twitchls.com; multitwitch.tv; etc; the playback works fine but on Twitch itself it doesn't. 

Any idea? Have tried with all the above user prefs but no luck.

**Comment by Thorin-Oakenpants** [comment]
Are you using Flash, or forcing the site to use HTML5?

**Comment by verlain3** [comment]
Nope, i don't use flash anymore. Not forcing html5 either by some sort of extension. 

I get a black screen and the pause/play button doesn't work, as well as the volume one. I'm not seeing what i'm possibly missing which makes it not playable on Twitch website but works fine on the other alternatives.

**Comment by verlain3** [comment]
Found the culprit.

2300: WEB WORKERS
change "user_pref("dom.workers.enabled", false);" to true made it work.

Any major downside be enabling this setting?

**Comment by Thorin-Oakenpants** [comment]
When it comes to problematic sites - eg twitch, or google docs, or outlook.com, or google maps, twitter, google street view - IMO people should use secondary profiles (or even a diff browser - eg I use Opera for a bunch of sites). Your default or MAIN FF profile shouldn't be compromised

That said - web workers will only become more common. The downside is it is not a per domain setting, so you open yourself up. There are some protections here AFAIK - workers must be same origin & scheme (and HTTPS only?). And FPI handles workers (but this seems weird given my previous sentence?). I also do not know the exact ins and outs of this, as previously I said (from documentation) that `service` workers require `web` workers - but in my test for IDB entries from youtube, that was not the case - so I guess the standards have/are changing or something is broken, or google are exploiting loopholes. The whole workers is quite frankly, a nightmare from a privacy standpoint.

At the end of the day it comes down to your browsing habits - either allow workers (2301) and/or use various sites in containers maybe - and accept the tradeoffs which will differ per person, eg logins used, etc. Or use a secondary profile.

**Comment by verlain3** [comment]
Thanks @Thorin-Oakenpants, always very informative. 

Cheers

**Comment by Atavic** [comment]
>Any major downside be enabling this setting?

Webworkers keep a connection a long time after you left a site like twitch. This way they eventually facilitate fingerprinting across teitch and the sites you visit there after.

**Comment by Atavic** [comment]
>Any major downside be enabling this setting?

Webworkers keep a connection a long time after you left a site like twitch. This way they eventually facilitate fingerprinting across twitch site and the one you visit thereafter.

**Comment by earthlng** [comment]
PSA: if you click on Popout Player you don't need workers enabled. A bit inconvenient but it works
